spec/eof.md

```diff
@@ -191,12 +176,16 @@ Code executing within an EOF environment will behave differently than legacy cod
 - `CREATE4 (0xed)` instruction
     - Works the same as `CREATE3` except:
         - does not have `initcontainer_index` immediate
-        - pops one more value from the stack (first argument): `tx_initcode_index`
-        - loads the initcode EOF container from the transaction `initcodes` array at `tx_initcode_index`
+        - pops one more value from the stack (first argument): `tx_initcode_hash`
+        - loads the initcode EOF container from the transaction `initcodes` array which hashes to `tx_initcode_hash`
             - fails (returns 0 on the stack) if such initcode does not exist in the transaction, including when there is no `initcodes` field at all
-                - this is a "light" failure: caller's nonce is not updated and gas for initcode execution is not consumed
-            - the initcode container and all its subcontainers are validated recursively. `CREATE4` fails (returns 0 on the stack) if any of these is invalid
-                - caller’s nonce is updated and gas for initcode execution is consumed
+            - fails (returns 0 on the stack) if `initcontainer_size > MAX_INITCODE_SIZE`
```

Comment:
I think this should be transaction validity rule and should not be checked during CREATE4.

Reply:
makes sense, and it is actually listed there :+1: